packages/jest-jasmine2/src/setup_jest_globals.ts

```diff
@@ -101,7 +101,7 @@ export default ({
 
   patchJasmine();
   const {expand, updateSnapshot} = globalConfig;
-  const snapshotResolver = buildSnapshotResolver(config);
+  const snapshotResolver = buildSnapshotResolver(config, localRequire);
```

Comment:
If `localRequire` is passed here, it won't fall back to the default of the parameter which does the transform, but instead just do a regular `requireInternalModule`, or am I missing something?

Reply:
It'll do whatever you pass in, which is the point - what's being passed in should take responsibility for handling the transformation, as that requires requiring the file to be transformed, which can't happen in an isolated module....

For the case of `Runtime`, that already includes doing transformation:
https://github.com/facebook/jest/blob/680c2f67bcac06f5d416a52dbbb1bc415501628e/packages/jest-runtime/src/index.ts#L723-L727

Reply:
But not if the module is marked internal, or does it? That's the part I don't understand

Reply:
The module being marked as internal is a separate thing.

My understanding is that this "cuts out" the transformation part, so that it's internal module state is independent on if `jest-snapshot` is being required as an internal module or not.

Reply:
Sorry, I'm dumb, mixed things up. `localRequire` in jasmine2/circus init is `runtime.requireModule`, not `runtime.requireInternalModule`